code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -172,6 +177,34 @@
 		// clear refs
 		controller = null
 
+/mob/living/critter/flock/drone/proc/release_control_abrupt()
+	src.flock?.hideAnnotations(src)
+	src.is_npc = TRUE
+	if(src.client && !controller)
+		controller = new/mob/living/intangible/flock/trace(src, src.flock)
+	if(!controller)
+		return
+	if (src.floorrunning)
+		src.end_floorrunning()
+		if (istype(src.loc, /turf/simulated/floor/feather))
+			var/turf/simulated/floor/feather/floor = src.loc
+			if (floor.on && !floor.connected)
+				floor.off()
```

Comment:
Don't floors already turn themselves off when `Exited()`? Or is this just paranoia

Reply:
This is done in released_control() too, Exited() will only be called for Flockdrones leaving the tile I think.